### DISCUSSION TITLE
Nuget packages compatibility

### DISCUSSION THREAD

**Discussion by SupermindPT** [dotnet/maui#15571] [Upgrading from Xamarin]
Hello everyone. 

I could not find anywhere this question, so I hope it was not posted already.

I have several Xamarin Forms projects, and I think at this time is probably good ideia to start migrating those projects to .NET MAUI. 

Most of these Xamarin Forms projects have lots of third party dependencies (some are very old nuget packages that are not updated for long time). My concern is that these APPs will not compile/ work properly because third party dependencies are not updated to support .NET 6 and .NET MAUI. 

Now I have read on .NET MAUI posts that developers should wait for those dependencies to support .NET MAUI, but what about the "deprecated" dependencies that are still used but will not be updated ? Are .NET 6 projects backward compatible with .NET Standard projects ?  

What should developers do in this situation ? 

Thank you in advance.

**Comment by acaliaro** [comment]
HI. Incompatible nugets are a problem. You could make a list of problematic packages and post it, so we can see if there are any alternatives.

**Reply by SupermindPT** [reply]
Here I list some nuget packages I think are incompatible with .NET MAUI:

- [Microcharts - Useful for drawing charts](https://www.nuget.org/packages/Microcharts/)
- [Refractored.FloatingActionButton - Useful for adding options to lists](https://www.nuget.org/packages/Refractored.FloatingActionButton/2.1.0?_src=template)
- [Refractored.FloatingActionButtonForms - Useful for adding options to lists](https://www.nuget.org/packages/Refractored.FloatingActionButtonForms/)
- [Xamarin.RisePlugin.AutoCompleteTextView](https://www.nuget.org/packages/Xamarin.RisePlugin.AutoCompleteTextView)
- [Matcha.BackgroundService - Good for background services](https://www.nuget.org/packages/Matcha.BackgroundService/)
- [TouchTracking.Forms](https://www.nuget.org/packages/TouchTracking.Forms/)
- [Xamanimation](https://www.nuget.org/packages/Xamanimation/)
- [SQLiteNetExtensions.Async - Async methods for SQLite and Cascade operations with Foreign Keys](https://www.nuget.org/packages/SQLiteNetExtensions.Async/2.1.0?_src=template)
- [Square.Picasso - I think I use this for image caching](https://www.nuget.org/packages/Square.Picasso/2.71828.0?_src=template)
- [Xamarin.Android.PhotoView](https://www.nuget.org/packages/Xamarin.Android.PhotoView/2.1.4?_src=template)
- [Xamarin.Google.Guava](https://www.nuget.org/packages/Xamarin.Google.Guava/)
- [PCLStorage - Useful for old Xamarin.Android projects](https://www.nuget.org/packages/PCLStorage/)

There are also other dependencies like Java Binding Libraries(.aar) and native Java libraries that have their own dependencies and that will probably stop working after the transition to .NET MAUI (or maybe not, now that I better think about it, those libraries are just native Java code). 
Seems like migrating to .NET MAUI is not the greatest ideia after all. Maybe in the next 2-3 years, .NET MAUI will keep up to Xamarin Forms current state (with regard to the nuget packages support).

**Reply by davidortinau** [reply]
Any of these that are open source on GitHub you can update to net6+ compat.

Microcharts can work with the existing NuGet and a workaround, but net6+ support has been added as well here https://github.com/microcharts-dotnet/Microcharts/pull/325 although the NuGet hasn't been published yet. @Eilon 

FAB - perhaps look at @VladislavAntonyuk's blog https://vladislavantonyuk.github.io/articles/Creating-dynamic-floating-action-button/

Xamanimation features are now in AlohaKit.Animations https://github.com/jsuarezruiz/AlohaKit.Animations

PCLStorage - I see several updated alternatives on NuGet.org. Can anyone confirm if any of those are suitable replacements?

Binding libraries work in .NET also, the csproj really just changes to SDK Style.

**Reply by SupermindPT** [reply]
I am starting to migrate some projects to .NET MAUI, starting with the smallest ones. We'll see how it goes. 
Thanks for the help!

**Comment by giuseppenovielli** [comment]
@SupermindPT Here there are some nuget packages migrated from Xamarin.Forms to .NET MAUI.
https://github.com/dotnet/maui/discussions/14459

**Reply by SupermindPT** [reply]
Thank you for the information

**Comment by 007vel** [comment]
PCLStorage - I see several updated alternatives on NuGet.org. Can anyone confirm if any of those are suitable replacements?

**Reply by filyXplor** [reply]
Any update on PCLStorage?
